### PR TITLE
Support pages output

### PR DESCRIPTION
**Pull request by tonykero**
I added an option to generate doxygen pages (which includes \page & \mainpage)
as requested in https://github.com/sourcey/moxygen/issues/14

i succesfully generated SFML Documentation and its pages
https://gist.github.com/tonykero/2dae639ef4b563e40d2a5e562b278f00#file-page-index-md

(ran with: `node ./moxygen-tonykero/bin/moxygen.js --pages --groups --output markdown/api-%s.md DOX/xml`)

> Note: title is not supported so there is a repeated "Welcome", maybe we could implement it as YAML metadata ?

### CODE REVIEW

**Comment by auscaster** [comment]
Great work on this, a really nice feature to have! Thank you

File: src/parser.js [hunk 1]
```diff
@@ -434,7 +434,7 @@ module.exports = {
 
       this.parseMembers(compound, element.$, element.member);
 
-      if (compound.kind !== 'page') { // && compound.kind !== 'file'
+      if (compound.kind !== 'file') { // && compound.kind !== 'file'
```
Comment (squeek502) [inline src/parser.js]: This change makes moxygen ignore xml files with 'file' root compounds, which breaks C parsing completely (unsure about how it affects C++).

I would submit a PR with a fix, but I'm unsure what the correct behavior is, or why this change was made over just removing the conditional entirely.
